spec/migrations/20190118204247_move_miq_set_memberships_spec.rb

```diff
@@ -0,0 +1,36 @@
+require_migration
+
+describe MoveMiqSetMemberships do
+  let(:rel_stub) { migration_stub(:Relationship) }
+  let(:membership_stub) { migration_stub(:MiqSetMembership) }
+  let(:ns_stub) { Class.new(ActiveRecord::Base) { self.table_name = "miq_ae_namespaces" } }
+  let(:all_set_memberships) { membership_stub.all }
+
+  migration_context :up do
+    context "parent/child/grandchild rel" do
+      it 'updates ancestry' do
+        ns1 = ns_stub.create!
+        ns1_child = ns_stub.create!
+        ns1_grandchild = ns_stub.create!
+        ns1_rel = rel_stub.create!(:relationship => "membership", :ancestry => nil, :resource_id => ns1.id, :resource_type => "MiqAeNamespace")
+        ns1_child_rel = rel_stub.create!(:relationship => "membership", :ancestry => ancestry_for(ns1_rel), :resource_id => ns1_child.id, :resource_type => "MiqAeNamespace")
+        rel_stub.create!(:relationship => "membership", :ancestry => ancestry_for(ns1_rel, ns1_child_rel), :resource_id => ns1_grandchild.id, :resource_type => "MiqAeNamespace")
+
+        migrate
+
+        membership = find_membership(ns1_child)
+        expect(membership.miq_set_id).to eq(ns1.id)
+        expect(membership.member_type).to eq("MiqAeNamespace")
+        expect(membership.member_id).to eq(ns1_child.id)
```

Comment:
would be nice to do a single test per membership.
not sure if it is `attributes includes(:miq_set_id => ns1.id, :member_type => ...)` that may read better?

Also, would be nice if more memberships are tested for here - the parent, child, and grand are all setup correctly.

 guess that is another reason why I'm trying to reduce the per object length - so you can test all 3 and it isn't too verbose - and the actual test will be lost in the boiler plate